src/keylessBackup/encryption.ts

```diff
@@ -0,0 +1,105 @@
+import * as secp from '@noble/secp256k1'
+import crypto from 'crypto'
+import hkdf from 'futoin-hkdf'
+
+/**
+ * Derives a 256-bit key using the HKDF method from two key shares.
+ *
+ * @param {string} keyshare1 - The first keyshare in utf8 format.
+ * @param {string} keyshare2 - The second keyshare in utf8 format.
+ * @returns {Buffer} A derived 256-bit key as a Buffer.
+ */
+export function deriveKeyFromKeyShares(
+  keyshare1: Buffer,
+  keyshare2: Buffer,
+  bytes: number = 32
+): Buffer {
+  // sanity check
+  if (keyshare1.length < 16 || keyshare2.length < 16) {
+    throw new Error('Key shares must be at least 16 bytes long')
+  }
+
+  // Combining the keyshares
+  const combinedKeyShares = Buffer.concat([
+    keyshare1,
+    hkdf(keyshare2, bytes, {
```

Comment:
I would recommend fixing this step to a length of 32. This gives the maximum available security available with SHA-256, and prevents a bit of wasted work if the requested output is longer than 32 bytes.
```suggestion
    hkdf(keyshare2, 32, {
```